File: backend/src/main/java/org/jdk/project/controller/config/TaskConfigController.java
package org.jdk.project.controller.config;

import lombok.RequiredArgsConstructor;
import org.jdk.project.dto.ListResponse;
import org.jdk.project.dto.config.TaskDto;
import org.jdk.project.dto.config.TaskUpsertRequest;
import org.jdk.project.service.ConfigCommandService;
import org.jdk.project.service.ConfigQueryService;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/config/tasks")
@RequiredArgsConstructor
public class TaskConfigController {

  private final ConfigCommandService configCommandService;
  private final ConfigQueryService configQueryService;

  @GetMapping
  public ListResponse<TaskDto> listTasks() {
    return configQueryService.listTaskTemplates();
  }

  @PostMapping
  public Long createTask(@RequestBody TaskUpsertRequest request) {
    return configCommandService.createTask(request);
  }

  @PutMapping("/{id}")
  public void updateTask(@PathVariable Long id, @RequestBody TaskUpsertRequest request) {
    configCommandService.updateTask(id, request);
  }

  @DeleteMapping("/{id}")
  public void deleteTask(@PathVariable Long id) {
    configCommandService.deleteTask(id);
  }

  @PostMapping("/{id}/run")
  public Long runTask(@PathVariable Long id) {
    return configCommandService.runTask(id);
  }
}

File: backend/src/main/kotlin/org/jdk/project/controller/config/TaskConfigController.kt
package org.jdk.project.controller.config

import org.jdk.project.dto.ListResponse
import org.jdk.project.dto.config.TaskDto
import org.jdk.project.dto.config.TaskUpsertRequest
import org.jdk.project.service.TaskService
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/config/tasks")
class TaskConfigController(
    private val taskService: TaskService,
) {
    @GetMapping
    fun listTaskTemplates(): ListResponse<TaskDto> = taskService.listTaskTemplates()

    @PostMapping
    fun createTask(
        @RequestBody request: TaskUpsertRequest,
    ): Long? = taskService.createTask(request)

    @PutMapping("/{id}")
    fun updateTask(
        @PathVariable id: Long,
        @RequestBody request: TaskUpsertRequest,
    ) {
        taskService.updateTask(id, request)
    }

    @DeleteMapping("/{id}")
    fun deleteTask(
        @PathVariable id: Long,
    ) {
        taskService.deleteTask(id)
    }

    @PostMapping("/{id}/run")
    fun runTask(
        @PathVariable id: Long,
    ): Long? = taskService.runTask(id)
}
